rust/connlib/libs/tunnel/src/lib.rs

```diff
@@ -504,6 +506,9 @@ where
                     // found some comments saying that a single read syscall represents a single packet but no docs on that
                     // See https://stackoverflow.com/questions/18461365/how-to-read-packet-by-packet-from-linux-tun-tap
                     match dev.device_channel.mtu().await {
+                        // XXX: Do we need to fetch the mtu every time? In most clients it'll
+                        // be hardcoded to 1280, and if not, it'll only change before packets start
+                        // to flow.
```

Comment:
I see. Pardon my noob question, but won't adding syscalls to the hotpath take up a lot of CPU time / overhead?

I ask because WireGuard-go [stores the MTU on the device struct](https://github.com/WireGuard/wireguard-go/blob/469159ecf7d108dc1d44f8789dedbb463b7bd3a6/device/device.go#L86), and boringtun [does something similar](https://github.com/cloudflare/boringtun/blob/e1d6360d6ab4529fc942a078e4c54df107abe2ba/boringtun/src/device/mod.rs#L382).

If so, I think it's an acceptable edge case for this to possibly crash if an admin changes the tunnel interface MTU while it's up.

Reply:
From there, if we do find this to be an unacceptable UX, we could refresh it periodically, or maybe it's possible to subscribe to the MTU set syscall?

Reply:
Now that I thought about this, I think that the `read` call will not fail if the buffer is small it will just read a mangled packet(the rest of this loop expects a full packet).

So we can just do as boringtun does that it reads the mtu once and updates it periodically. Still calling a method here is fine, the `DeviceChannel` can store the mtu.

Reply:
Gotcha, that makes sense. Thanks for clarifying! Glad it wasn't a stupid question.